torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -0,0 +1,160 @@
+        r"""
+        The purpose of this method is to provide a user the set of all module_fqns so that if
+        they wish to use some of the filtering capabilities of the ModelReportVisualizer class,
+        they don't need to manually parse the generated_reports dictionary to get this information.
+
+        Returns all the unique module fqns present in the reports the ModelReportVisualizer
+        instance was initialized with.
+        """
+        pass
+
+    def get_all_unique_feature_names(self, plottable: bool) -> Set[str]:
+        r"""
+        The purpose of this method is to provide a user the set of all feature names so that if
+        they wish to use the filtering capabilities of the generate_table_view(), or use either of
+        the generate_plot_view() or generate_histogram_view(), they don't need to manually parse
+        the generated_reports dictionary to get this information.
+
+        Args:
+            plottable (bool): True if the user is only looking for plottable features, False otherwise
+                plottable features are those that are tensor values
+
+        Returns all the unique module fqns present in the reports the ModelReportVisualizer
+        instance was initialized with.
+        """
+        pass
+
+    def generate_table_view(self, feature: str = "", module_fqn_prefix_filter: str = "") -> Tuple[List[List[Any]], str]:
+        r"""
+        Takes in optional filter values and generates a table with the desired information.
+
+        The generated table is presented in both a list-of-lists format for further manipulation and filtering
+        as well as a formatted string that is ready to print
+
+        Table columns:
+
+         idx  layer_fqn   type  shape  feature_1   feature_2   feature_3   .... feature_n
+        ----  ---------   ----  -----  ---------   ---------   ---------        ---------
+
+        Args:
+            feature (str, optional): The specific feature we wish to generate the table for
+                Default = "", results in all the features being printed
+            module_fqn_prefix_filter (str, optional): Only includes modules with this string prefix
+                Default = "", results in all the modules in the reports to be visible in the table
+
+        Returns a tuple with two objects:
+            (List[List[Any]]) A list of lists containing the table information row by row
+                The 0th index row will contain the headers of the columns
+                The rest of the rows will contain data
+            (str) The formatted string that contains the table information to be printed
+        Expected Use:
+            >>> info, tabluated_str = model_report_visualizer.generate_table_view(*filters)
+            >>> print(tabulated_str) # outputs neatly formatted table
+        """
+        pass
+
+    def generate_plot_view(self, feature: str, module_fqn_prefix_filter: str = "") -> List[List[Any]]:
```

Comment:
Why does `generate_table_view` return an info string but not `generate_plot_view` and `generate_histogram_view`? I feel the return signatures of all these should be the same

Reply:
the thing that all three return consistently is the list of lists representing the filtered data for them to go through. The reason `generate_table_view` returns a string is because we want to present the table in a string format so they can just print it see [this line](https://github.com/pytorch/pytorch/pull/81523/files/92227b0ffa3eb62d878679af8f60e67bd282689f#diff-5d4780a142975c963b00de031975b2b538df9ff8a069d3b43d7c4099165fb5dfR113) for expected usage.

Meanwhile, **I agree about the inconsistency with the histogram and plot, so I update those to be in the same tuple** format, but they return functions that can be used plot the graphs instead of strings to print